server/server.go

```diff
@@ -0,0 +1,61 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/brave-intl/bat-go/middleware"
+	"github.com/brave/go-translate/controller"
+	"github.com/getsentry/raven-go"
+	"github.com/go-chi/chi"
+	chiware "github.com/go-chi/chi/middleware"
+	"github.com/pressly/lg"
+	"github.com/sirupsen/logrus"
+)
+
+func setupLogger(ctx context.Context) (context.Context, *logrus.Logger) {
+	logger := logrus.New()
+	// Redirect output from the standard logging package "log"
+	lg.RedirectStdlogOutput(logger)
+	lg.DefaultLogger = logger
+	ctx = lg.WithLoggerContext(ctx, logger)
+	return ctx, logger
+}
+
+func setupRouter(ctx context.Context, logger *logrus.Logger) (context.Context, *chi.Mux) {
+	r := chi.NewRouter()
+
+	r.Use(chiware.RequestID)
+	r.Use(chiware.RealIP)
```

Comment:
@yrliou - Why do we need RealIP, looks like it sets the `X-Forwarded-*` headers? https://godoc.org/github.com/go-chi/chi/middleware#RealIP

Reply:
I think this is for requests coming in from brave-core, but not the requests we're sending out to Microsoft AFAIK.
And based on the doc, it's setting the remote address in the requests instead of setting the header.
It's a shared pattern from bat-go and go-update btw.

Reply:
My concern was not about the headers sent to Microsoft, but more in terms of adding more avenues to log user ips on the go-translate server. For now, having a log-retention policy on the infrastructure side sounds like a good idea.